cassdegrees/ui/forms.py

```diff
@@ -0,0 +1,110 @@
+import json
+
+from api.models import DegreeModel, SubplanModel
+from django import forms
+from django.core.exceptions import NON_FIELD_ERRORS
+from django.forms import ModelForm
+
+
+class JSONField(forms.CharField):
+
+    def __init__(self, *args, field_id, **kwargs):
+        super().__init__(widget=forms.HiddenInput(attrs={'id': field_id}), *args, **kwargs)
+
+    def to_python(self, value):
+        return json.loads(value)
+
+
+class EditProgramFormSnippet(ModelForm):
+    # Use custom handlers for JSON fields
+    globalRequirements = JSONField(field_id='globalRequirements', required=False)
+
+    class Meta:
+        model = DegreeModel
+        fields = ('code', 'year', 'name', 'units', 'degreeType', 'globalRequirements')
+        widgets = {
+            'code': forms.TextInput(attrs={'class': "text tfull"}),
+            'year': forms.NumberInput(attrs={'class': "text tfull", 'type': "number"}),
+            'name': forms.TextInput(attrs={'class': "text tfull"}),
```

Comment:
```suggestion
            'name': forms.TextInput(attrs={'class': "text tfull", 'placeholder': "e.g. Bachelor of Arts"}),
```

Adds sample placeholder text to Program forms